src/foam/u2/filter/properties/PropertyFilterView.js

```diff
@@ -182,6 +187,33 @@ foam.CLASS({
     },
 
     function isFiltering() {
+      if ( ! this.isInit ) {
+        if ( ! this.view_ )
+          return;
+
+        if ( ! this.memento.paramsObj.filters ) {
+          this.memento.paramsObj.filters = [];
+        }
+
+        this.memento.paramsObj.filters = this.memento.paramsObj.filters.filter(f => f.name !== this.property.name && f.criteria !== this.criteria );
+
+        var pred;
+        if ( Object.keys(this.view_.predicate).length > 0 && ! foam.mlang.predicate.True.isInstance(this.view_.predicate) )
+          pred =  foam.json.Outputter.create({
+            strict: true
+          }).stringify(this.view_.predicate);
```

Comment:
does stringifying predicates creates security issue?

Reply:
@kgrgreer we recreate back into an MLang java object and we don't 'eval' the json string, so this should not be an issue.

Reply:
I'm not super keen on using JSON for mementos because it doesn't give very natural looking URL's, but we can address that in a future version.